### PR TITLE
Mccalluc/use saucelabs

### PR DESCRIPTION
**Pull request by mccalluc**
Tests pass on Saucelabs! @sgratzl : Are there blocking issues here before we can merge?

I'll continue to add tests incrementally, but I would really like to get this back into master.

### CODE REVIEW

**Comment by mccalluc** [comment]
(On the last commit, one passed, and the other failed because we ran out of saucelab tunnels. Perhaps just do one build on each commit? Not sure what is best.)

**Comment by mccalluc** [comment]
I've been flying blind on this project, and one of the things I've been uncertain of is whether the methods I might be trying to test are really in use. There are a range of possibilities:
- They should be part of the public API, and existing applications depend on them.
- They should be part of the public API, but we don't have any examples of them being used.
- They are used in Core, but aren't really part of the public API. (There isn't a good way to hide these, though there are proposals: https://github.com/Microsoft/TypeScript/issues/5228)
- They are used within the module, but could be made private.
- Or, maybe they are never used after being defined.

I'm taking a semi-automated approach to identifying these, but as always I could use feedback. Thanks!

File: src/main.ts [hunk 1]
```diff
@@ -173,7 +171,7 @@ export function constantTrue() {
  * @returns {boolean}
  */
 export function constantFalse() {
-  return true;
```
Comment (thinkh) [inline src/main.ts:176]: Interesting mistake. ;-)

**Comment by thinkh** [comment]
@mccalluc You are right, you cannot know which function is used and which not. I would go for the naive approach and clone all repos we have and then perform a global search (i.e. over all projects) with your text editor for each public/exported function. Maybe you have an idea if and how to automate these steps. 
In my opinion, if the function isn't used anywhere, you can leave a comment and mark it as private (or remove the export). 
Other than that you don't have to care much about other plugins. And of course you don't have to test private functions. ;-) Otherwise we will never finish this topic and move on.

**Comment by ngehlenborg** [comment]
@mccalluc: is this PR still current?

**Comment by mccalluc** [comment]
I won't be merging it, but I should translate the tests from js to ts. I think the tests that just check the list of properties also served a role, but I'm less clear about them going forward.

**Comment by mccalluc** [comment]
I copied all these tests to a new branch, and began translating them to the new syntax in #12. There is more translation work to be done, but it has all been merged to master and there's no reason to keep this around.

**Pull request closed without merging**